examples/communication/child_to_parent/src/main.rs

```diff
@@ -0,0 +1,86 @@
+use yew::{html, AttrValue, Callback, Component, Context, Html, Properties};
+
+// The `Parent` component is the parent of the `Child` component.
+
+pub struct Parent {
+    value: u32,
+}
+
+pub enum Msg {
+    ButtonClick,
+}
+
+impl Component for Parent {
+    type Message = Msg;
+    type Properties = ();
+
+    fn create(_ctx: &Context<Self>) -> Self {
+        Self { value: 0 }
+    }
+
+    fn update(&mut self, _ctx: &Context<Self>, msg: Self::Message) -> bool {
+        match msg {
+            Msg::ButtonClick => {
+                self.value += 1;
+                true
+            }
+        }
+    }
+
+    fn view(&self, ctx: &Context<Self>) -> Html {
+        let msg = format!("My children have been clicked {} times", self.value);
+        let on_clicked = ctx.link().callback(|_| Msg::ButtonClick);
+        html! {
+            <div class="app">
+                <div class="parent">
+                    <h2>{ "Child-to-Parent communication example" }</h2>
+                    <div>{msg}</div>
+                    <Child name="Alice" on_clicked={on_clicked.clone()} />
+                    <Child name="Bob" {on_clicked} />
+                </div>
+            </div>
+        }
+    }
+}
+
+// The `Child` component is the child of the `Parent` component, and will send updates to the parent using a Callback.
+
+pub struct Child;
+
+#[derive(Clone, PartialEq, Properties)]
+pub struct ChildProps {
+    pub name: AttrValue,
+    pub on_clicked: Callback<()>,
+}
+
+impl Component for Child {
+    type Message = ();
+    type Properties = ChildProps;
+
+    fn create(_ctx: &Context<Self>) -> Self {
+        Self {}
+    }
+
+    fn view(&self, ctx: &Context<Self>) -> Html {
+        let name = format!("{}:", ctx.props().name);
+
+        // Here we get a reference to the callback set by the parent component.
+        let on_clicked = ctx.props().on_clicked.clone();
+
+        // Here we emit the callback to the parent component, whenever the button is clicked.
+        let onclick = move |_| on_clicked.emit(());
```

Comment:
`let onclick = ctx.props().on_clicked.reform(|_| ())` would be more concise and more to the point that all we're doing here is ignore the exact event, instead providing `()` to the passed in callback.

Reply:
Nice! Learned something new today. :-)